include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,519 @@
+  {
+    using namespace dealii;
+
+    namespace internal
+     {
+       namespace
+       {
+         bool
+         fexists(const std::string &filename)
+         {
+           std::ifstream ifile(filename.c_str());
+           return !(!ifile); // only in c++11 you can convert to bool directly
+         }
+       }
+
+       template <int dim, int grid_dim>
+       class AsciiDataLookup
+       {
+         public:
+           AsciiDataLookup(const std_cxx11::array<unsigned int,3> &npoints,
+                           const GeometryModel::Interface<dim> &geometry_model,
+                           const unsigned int components,
+                           const double scale_factor,
+                           const types::boundary_id boundary_id = numbers::invalid_boundary_id)
+         :
+           components(components),
+           geometry_model(geometry_model),
+           data(components),
+           old_data(components),
+           scale_factor(scale_factor)
+         {
+             Assert (grid_dim <= dim,
+                 ExcMessage("You can only create an AsciiDataLookup object for a grid of at "
+                     "most the number of dimensions your model uses."));
+
+             const std_cxx11::array<unsigned int,grid_dim> dimensions = get_boundary_dimensions(boundary_id);
+             const std_cxx11::array<std::pair<double,double>,dim> model_extent = get_model_extent(geometry_model);
+
+             for (unsigned int i = 0; i < grid_dim; i++)
+               {
+                 boundary_dimensions[i] = dimensions[i];
+                 grid_extent[i] = model_extent[boundary_dimensions[i]];
+                 table_points[i] = npoints[boundary_dimensions[i]];
+               }
+         }
+
+           void screen_output(const ConditionalOStream &pcout) const
+           {
+             std::ostringstream output;
+
+             output << std::setprecision (3) << std::setw(3) << std::fixed << std::endl
+                 << "   Set up Ascii Data boundary module."  << std::endl
+                 << "   The grid extents to:";
+             for (unsigned int i = 0; i < grid_dim; i++)
+               output << " " << grid_extent[i].second;
+
+             output << std::endl;
+
+             pcout << output.str();
+           }
+
+           void
+           load_file(const std::string &filename)
+           {
+             // Check whether file exists, we do not want to throw
+             // an exception in case it does not, because it could be by purpose
+             // (i.e. the end of the boundary condition is reached)
+             AssertThrow (fexists(filename),
+                 ExcMessage (std::string("Ascii data file <")
+             +
+             filename
+             +
+             "> not found!"));
+
+             std::ifstream in(filename.c_str(), std::ios::in);
+             AssertThrow (in,
+                          ExcMessage (std::string("Couldn't find data. Is file format correct?")));
```

Comment:
That's the wrong error message -- it's not that the file format is incorrect, but that the file couldn't be read. Or do I misread? I mean, how can this assertion fail fi the one above succeeds?